src/client/pkg/tls/cert_loader.go

```diff
@@ -0,0 +1,79 @@
+package tls
+
+import (
+	"crypto/tls"
+	"fmt"
+	"sync/atomic"
+	"time"
+	"unsafe"
+
+	"github.com/pachyderm/pachyderm/src/client/pkg/errors"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// CertLoader provides simple hot TLS certificate reloading by checking for a renewed certificate at a configurable interval
+type CertLoader struct {
+	certPath        string
+	keyPath         string
+	refreshInterval time.Duration
+
+	// cert is the current cached *tls.Certificate. It should only be accessed with atomic methods because it may be updated by the cert reloading routine.
+	cert     unsafe.Pointer
```

Comment:
What's the trade-off of atomic + unsafe pointers vs a plain old mutex?

Reply:
My intution was atomic ops would introduce less latency and the implementation is still pretty simple (and encapsulated so users aren't exposed to unsafe). I don't have any data to back that up though :) It could be implemented with an RWMutex and I doubt anyone would notice the difference.